src/MiddlewareController.php

```diff
@@ -26,6 +30,12 @@
 
 /**
  * @internal
+ *
+ * @property PluginManager         $plugins
+ * @property RequestInterface      $request
+ * @property ResponseInterface     $response
+ * @property EventManagerInterface $event
+ * @property EventInterface        $events
```

Comment:
I don't understand why those are needed but $event and $events types are swapped here. Besides, $event is a `MvcEvent`